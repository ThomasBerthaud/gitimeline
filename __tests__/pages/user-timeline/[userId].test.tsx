import { describe, expect, test } from 'vitest';
import { render, screen } from '@testing-library/react';
import UserTimeline, { getServerSideProps } from '../../../pages/user-timeline/[userId]';
import { GetServerSidePropsContext } from 'next';
import { userRepositories } from '../../mocks/handlers';

describe('user timeline page', () => {
  test("should display user's  timeline", () => {
    const { asFragment } = render(<UserTimeline userRepositories={userRepositories} />);

    expect(screen.getAllByRole('heading', { level: 4 })).toHaveLength(2);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should fetch api data', async () => {
    const context = { params: { userId: 'userId' } } as GetServerSidePropsContext<{ userId: string }>;
    const result = await getServerSideProps(context);
    expect(result).toEqual({ props: { userRepositories } });
  });
});
