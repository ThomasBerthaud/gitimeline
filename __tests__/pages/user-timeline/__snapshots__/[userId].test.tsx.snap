// Vitest Snapshot v1

exports[`user timeline page > should display user's  timeline 1`] = `
<DocumentFragment>
  <div
    class="my-3"
  >
    <h3
      class="text-lg font-bold my-2"
    >
      user
    </h3>
    <div
      class="m-2"
    >
      <div
        class="bg-amber-50 rounded-full border-2 border-amber-200 w-4 h-4 m-2"
      />
      <h4
        class="text-lg font-bold"
      >
        repo
      </h4>
      <a
        class="link text-xs mx-1"
        href="repoUrl"
      >
        (repoUrl)
      </a>
      <div
        class="text-italic"
      >
        2021-07-11T18:45:06Z
      </div>
    </div>
    <div
      class="m-2"
    >
      <div
        class="bg-amber-50 rounded-full border-2 border-amber-200 w-4 h-4 m-2"
      />
      <h4
        class="text-lg font-bold"
      >
        repo2
      </h4>
      <a
        class="link text-xs mx-1"
        href="repoUrl2"
      >
        (repoUrl2)
      </a>
      <div
        class="text-italic"
      >
        2022-07-11T18:45:06Z
      </div>
    </div>
  </div>
</DocumentFragment>
`;
